go/tasks/plugins/k8s/ray/ray.go

```diff
@@ -0,0 +1,353 @@
+package ray
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins"
+	"github.com/flyteorg/flyteplugins/go/tasks/errors"
+	"github.com/flyteorg/flyteplugins/go/tasks/logs"
+	"github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery"
+	pluginsCore "github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery/flytek8s"
+	"github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery/k8s"
+	"github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery/utils"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	v1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	rayTaskType = "ray"
+	KindRayJob  = "RayJob"
+)
+
+type rayJobResourceHandler struct {
+}
+
+func (rayJobResourceHandler) GetProperties() k8s.PluginProperties {
+	return k8s.PluginProperties{}
+}
+
+// BuildResource Creates a new ray job resource.
+func (rayJobResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCore.TaskExecutionContext) (client.Object, error) {
+	taskTemplate, err := taskCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return nil, errors.Errorf(errors.BadTaskSpecification, "unable to fetch task specification [%v]", err.Error())
+	} else if taskTemplate == nil {
+		return nil, errors.Errorf(errors.BadTaskSpecification, "nil task specification")
+	}
+
+	rayJob := plugins.RayJob{}
+	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &rayJob)
+	if err != nil {
+		return nil, errors.Errorf(errors.BadTaskSpecification, "invalid TaskSpecification [%v], Err: [%v]", taskTemplate.GetCustom(), err.Error())
+	}
+
+	podSpec, err := flytek8s.ToK8sPodSpec(ctx, taskCtx)
+	if err != nil {
+		return nil, errors.Errorf(errors.BadTaskSpecification, "Unable to create pod spec: [%v]", err.Error())
+	}
+
+	headReplicas := int32(1)
+	headNodeRayStartParams := make(map[string]string)
+	if rayJob.RayCluster.HeadGroupSpec != nil && rayJob.RayCluster.HeadGroupSpec.RayStartParams != nil {
+		headNodeRayStartParams = rayJob.RayCluster.HeadGroupSpec.RayStartParams
+	}
+	headNodeRayStartParams["include-dashboard"] = "true"
+	headNodeRayStartParams["node-ip-address"] = "$MY_POD_IP"
+	headNodeRayStartParams["dashboard-host"] = "0.0.0.0"
```

Comment:
This means that "include-dashboard" is always true because it overrides the RayCluster parameters set by the user right? Should this be user-configurable?

Reply:
Updated it. yes, it should be user-configurable. If users don't set this value in flytekit, will use the default value in propeller configMap.